### ISSUE TITLE
Entire script does not work when "Add button to toggle to the currently embedded video's chat" is enabled

### ISSUE THREAD

**Issue by Mojavve** [vyneer/dgg-chat-gui-scripts#64]
```
Browser: Firefox Nightly 112.0a1 (2023-02-16) (64-bit)
Userscript version: d.gg utilities 1.7.3
Userscript plugin: Tampermonkey 4.18.1
```

Today I noticed DGG Utilities wasn't working.  It didn't show embeds, the embed button was there however clicking it did nothing.  The "d.gg utilities settings" button appeared under settings but clicking it led to a blank popup.  I tried a number of things and what eventually led to fixing it was opening the data.sqlite database for destiny.gg and clearing out the saved vyneer settings.  So I set my settings up again and upon next refresh, it failed to work again.  So instead of clearing the saved settings **I switched "vyneer-util.embedChat" to "false" and the script worked again.  Did this a few more times to make sure that was it, and yup it being enabled completely screws up the entire script.**

When it is screwed up, here are the errors I get in the devtools console:

```
Uncaught TypeError: can't access property "id", hostInfo is null
    isHost moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:716
    getEmbedChatToggleButtonText moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:764
    addEmbedChatToggleBtn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:846
    injectScript moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:892
    EventListener.handleEvent*window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:220
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    "__f__le85pupj.31t" moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    n https://www.destiny.gg/embed/chat?follow=/bigscreen:71
    <anonymous> https://www.destiny.gg/embed/chat?follow=/bigscreen:72
    set https://www.destiny.gg/embed/chat?follow=/bigscreen:81
    <anonymous> moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
eval:716:5
    isHost moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:716
    getEmbedChatToggleButtonText moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:764
    addEmbedChatToggleBtn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:846
    injectScript moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:892
    (Async: EventListener.handleEvent)
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:220
    call self-hosted:1497
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    "__f__le85pupj.31t" moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    call self-hosted:1497
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    n https://www.destiny.gg/embed/chat?follow=/bigscreen:71
    <anonymous> https://www.destiny.gg/embed/chat?follow=/bigscreen:72
    set https://www.destiny.gg/embed/chat?follow=/bigscreen:81
    <anonymous> moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    _o moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:21
    O moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:41
    Lo moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:42
    forEach self-hosted:203
    call self-hosted:1497
    Mt moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:8
    t moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:9
    Lo moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:38
    Ko moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:54
    Go moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:53
    Sn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:15
    forEach self-hosted:203
    call self-hosted:1497
    Mt moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:8
    t moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:9
    Sn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:15
    s moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:50
    apply self-hosted:2871
    applySafeWithoutClone resource://gre/modules/ExtensionCommon.jsm:641
    applySafeWithoutClone self-hosted:1497
    wrapPromise resource://gre/modules/ExtensionCommon.jsm:905
```

```
Uncaught ReferenceError: can't access lexical declaration 'DGGMsg' before initialization
    embeds moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:2473
    updateObserver moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:373
    injectScript moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:331
    EventListener.handleEvent*window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:220
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    "__f__le85pupj.31t" moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    n https://www.destiny.gg/embed/chat?follow=/bigscreen:71
    <anonymous> https://www.destiny.gg/embed/chat?follow=/bigscreen:72
    set https://www.destiny.gg/embed/chat?follow=/bigscreen:81
    <anonymous> moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
eval:2473:7
    embeds moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:2473
    updateObserver moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:373
    (Async: MutationCallback)
    injectScript moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:331
    (Async: EventListener.handleEvent)
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:220
    call self-hosted:1497
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    window["__f__le85pupj.31t"]/< moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    "__f__le85pupj.31t" moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    call self-hosted:1497
    St https://www.destiny.gg/embed/chat?follow=/bigscreen:9
    n https://www.destiny.gg/embed/chat?follow=/bigscreen:71
    <anonymous> https://www.destiny.gg/embed/chat?follow=/bigscreen:72
    set https://www.destiny.gg/embed/chat?follow=/bigscreen:81
    <anonymous> moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/userscripts/d.gg utilities.user.js?id=781429b6-002d-4e17-8f18-f45f21352aa4:1
    _o moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:21
    O moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:41
    Lo moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:42
    forEach self-hosted:203
    call self-hosted:1497
    Mt moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:8
    t moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:9
    Lo moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:38
    Ko moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:54
    Go moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:53
    Sn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:15
    forEach self-hosted:203
    call self-hosted:1497
    Mt moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:8
    t moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:9
    Sn moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:15
    s moz-extension://a07a54f5-b122-40ec-83be-1a5b47775b5b/content.js:50
    apply self-hosted:2871
    applySafeWithoutClone resource://gre/modules/ExtensionCommon.jsm:641
    applySafeWithoutClone self-hosted:1497
    wrapPromise resource://gre/modules/ExtensionCommon.jsm:905

```

**Comment by mattroseman** [comment]
Hey, I'm not a maintainer, but I wrote the code that's causing this issue.
I have a [PR](https://github.com/vyneer/dgg-chat-gui-scripts/pull/65) to fix it, but like you mention, in the meantime you can:

(1) Run this command in your browser's console to disable the bugged code:
`localStorage.setItem('vyneer-util.embedChat', false);`

(2) Open just the chat at [chat.destiny.gg](https://chat.destiny.gg), go to the plugin's settings, and disable the "Add button to toggle to the currently embedded video's chat" option.
<img width="545" alt="Screenshot 2023-02-17 at 1 46 52 AM" src="https://user-images.githubusercontent.com/6654122/219570128-7e912498-911a-47da-82e8-2c642914448a.png">

**Comment by vyneer** [comment]
Fixed in 4669e2e14c4772ff1743a4945d57642555952396, huge thanks to @mattroseman as usual ❤️
